### PR TITLE
drivers: wifi: Update RPU patch to rev#f52e25e1aeb

### PR DESCRIPTION
**Pull request by karun2796**
[SHEL-3331] : UMAC Tx Packet injector mode changes.

### CODE REVIEW

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 1]
```diff
@@ -1050,6 +1066,191 @@ struct nrf_wifi_cmd_get_stats {
...
+	/** TX injection mode setting enable */
+	NRF_WIFI_TX_INJECTION_MODE = 0x4,
+	/** Promiscuous mode setting enable */
+	NRF_WIFI_PROMISCUOUS_MODE = 0x8,
+	/** AP mode setting enable */
+	NRF_WIFI_AP_MODE = 0x10,
+	/** STA-AP mode setting enable */
+	NRF_WIFI_STA_AP_MODE = 0x20,
+	/** Max limit check based on current modes supported*/
+	WIFI_MODE_LIMIT_CHK = x02f,
```
Comment (VivekUppunda) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1096]: This should be 0x2f i think

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 2]
```diff
@@ -1050,6 +1066,191 @@ struct nrf_wifi_cmd_get_stats {
...
 } __NRF_WIFI_PKD;
 
+/**
+ * @brief This structure represents the channel parameters to configure specific channel.
+ *
+ */
+struct nrf_wifi_cmd_set_channel {
+	/** UMAC header, @ref nrf_wifi_sys_head*/
+	struct nrf_wifi_sys_head sys_head;
+	/* Interface index  */
```
Comment (VivekUppunda) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1076]: probably check the doxygen style in this document

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h (hunk 1 above)
Comment (karun2796) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h]: Taken care

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h (hunk 2 above)
Comment (karun2796) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1076]: Taken care.

**Review by VivekUppunda — APPROVED** [review]

**Review by udaynordic — APPROVED** [review]

**Review by rajb9 — APPROVED** [review]

**Pull request closed without merging**